File: app/src/main/java/com/example/kirppis/Item.java
package com.example.kirppis;
import com.google.firebase.database.IgnoreExtraProperties;

@IgnoreExtraProperties
public class Item {

    private String name;
    private String price;
    private String description;
    private String image;

    public Item() {
        // Default constructor required for calls to DataSnapshot.getValue(User.class)
    }

    public Item(String dbName, String dbPrice, String dbDescription, String dbImage) {
        name = dbName;
        price = dbPrice;
        description = dbDescription;
        image = dbImage;
    }

    public String getName() {
        return name;
    }

    public String getPrice() {
        return price;
    }

    public String getDescription() {
        return description;
    }

    public String getImage() {
        return image;
    }
}

File: app/src/main/java/com/example/kirppis/Item.kt
package com.example.kirppis

import com.google.firebase.database.IgnoreExtraProperties

@IgnoreExtraProperties
class Item {
    var name: String? = null
        private set
    var price: String? = null
        private set
    var description: String? = null
        private set
    var image: String? = null
        private set

    constructor() {
        // Default constructor required for calls to DataSnapshot.getValue(User.class)
    }

    constructor(dbName: String?, dbPrice: String?, dbDescription: String?, dbImage: String?) {
        name = dbName
        price = dbPrice
        description = dbDescription
        image = dbImage
    }
}